testcase/unittest/test_tx_message.py

```diff
@@ -0,0 +1,166 @@
+import os
+import random
+import threading
+from typing import List, Tuple, Callable
+
+import pytest
+import time
+
+from loopchain import configure
+from loopchain.baseservice.tx_message import TxItem, TxMessages, TxMessagesQueue
+from loopchain.blockchain import ExternalAddress, Transaction
+from loopchain.blockchain.transactions import TransactionVersioner, TransactionBuilder
+from loopchain.crypto.signature import Signer
+
+
+@pytest.fixture
+def tx_item() -> Callable[[TransactionVersioner], Transaction]:
+    def _tx_item(tx_versioner: TransactionVersioner) -> Transaction:
+        test_signer = Signer.from_prikey(os.urandom(32))
+        tx_builder = TransactionBuilder.new("0x3", "", tx_versioner)
+        tx_builder.signer = test_signer
+        tx_builder.to_address = ExternalAddress.new()
+        tx_builder.step_limit = random.randint(0, 10000)
+        tx_builder.value = random.randint(0, 10000)
+        tx_builder.nid = 2
+        tx: Transaction = tx_builder.build()
+        return tx
+    return _tx_item
+
+
+@pytest.fixture
+def tx_items(request, tx_item) -> List[Tuple[Transaction, TransactionVersioner]]:
+    tx_versioner = TransactionVersioner()
+    tx_params = []
+
+    tx_count = request.param
+    print(f"tx_count : {request.param}")
+
+    for i in range(tx_count):
+        tx: Transaction = tx_item(tx_versioner)
+        # print(f"tx : {tx.raw_data}")
+        tx_params.append((tx, tx_versioner))
+
+    return tx_params
+
+
+class TestTxMessages:
+    tx_count = 50
+
+    @pytest.mark.parametrize("tx_items", [tx_count], indirect=True)
+    def test_tx_messages(self, tx_items: List[Tuple[Transaction, TransactionVersioner]]):
+
+        # given
+        tx_size = 0
+        tx_messages = TxMessages()
+        for tx_param in tx_items:
+            tx_item = TxItem.create_tx_item(tx_param, 'icon_dex')
+            tx_messages.append(tx_item)
+            tx_size += len(tx_item)
+
+        # when
+        tx_total_size = tx_messages.total_size
+        tx_total_count = len(tx_messages)
+
+        # then
+        assert tx_total_count == self.tx_count
+        assert tx_total_size == tx_size
+
+
+class TestTxMessagesQueue:
+    tx_count = 50
+
+    def test_custom_tx_value(self):
+        print()
+        # given custom max_tx_size and max_tx_count
+        given_max_tx_size = 1024
+        given_max_tx_count = 100
+
+        configure.MAX_TX_SIZE_IN_BLOCK = given_max_tx_size
+        configure.MAX_TX_COUNT_IN_ADDTX_LIST = given_max_tx_count
```

Comment:
Please consider monkeypatch global variables: 
```python
        mocker.patch.object(configure, "MAX_TX_SIZE_IN_BLOCK", given_max_tx_size)
        mocker.patch.object(configure, "MAX_TX_COUNT_IN_ADDTX_LIST", given_max_tx_count)
```

You try to change global variables. It might be effective to other tests which uses those variables.
Try to print them out in another tests if they are changed or not.

Reply:
updated